lib/ansible/modules/remote_management/manageiq/manageiq_provider.py

```diff
@@ -464,26 +459,28 @@ def create_provider(self, name, provider_type, endpoints, zone_id, provider_regi
 def main():
     zone_id = None
     endpoints = []
+    argument_spec = dict(
+        manageiq_connection=dict(required=True, type='dict',
+                                 options=manageiq_argument_spec()),
+        state=dict(choices=['absent', 'present'], default='present'),
+        name=dict(required=True),
+        zone=dict(default='default'),
+        provider_region=dict(),
+        type=dict(choices=supported_providers().keys()),
+    )
+    # add the endpoint arguments to the arguments
+    argument_spec.update(endpoint_list_spec())
 
     module = AnsibleModule(
-        argument_spec=dict(
-            manageiq_connection=dict(required=True, type='dict',
-                                     options=manageiq_argument_spec()),
-            state=dict(choices=['absent', 'present'], default='present'),
-            name=dict(required=True),
-            zone=dict(default='default'),
-            provider_region=dict(),
-            type=dict(choices=supported_providers().keys()),
-            endpoints=dict(type='dict', options=endpoint_list_spec()),
-        ),
+        argument_spec=argument_spec,
         required_if=[
-            ('state', 'present', ['endpoints'])],
+            ('state', 'present', ['provider'])],
     )
 
     name = module.params['name']
     zone_name = module.params['zone']
     provider_type = module.params['type']
-    raw_endpoints = module.params['endpoints']
+    raw_endpoints = module.params
```

Comment:
(nit, your call) this is not just endpoints but all params.  that works fine, but consider dropping this var and directly using `module.params`.

Reply:
👍 just less things to change :-)